svcb.go

```diff
@@ -667,8 +661,9 @@ func (s *SVCBECHConfig) copy() SVCBKeyValue {
 	return &SVCBECHConfig{cloneSlice(s.ECH)}
 }
 
-func (s *SVCBECHConfig) unpack(b []byte) error {
-	s.ECH = cloneSlice(b)
+func (s *SVCBECHConfig) unpack(b *cryptobyte.String) error {
+	s.ECH = cloneSlice(*b)
+	b.Skip(len(*b))
```

Comment:
slightly concerning that a b.Skip is needed here. Should there be a cloneSliceAndSkip function?

i.e. it could be nice if these become 1-liners? :
~~~ go
func (s *SVCBECHConfig) unpack(b *cryptobyte.String) error { s.ECH = clonseSliceAndSkip(*b); return nil}
~~~